### PR TITLE
Feature/resolver

### PR DESCRIPTION
**Pull request by JamesDalboth**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Description
To help with versioning we are going to use ENS which is like DNS but for ethereum, am currently in the process of buying a domain for that, we can then link a domain to a resolver which I have written below which we can use to upload new versions from backend and the front end can easily switch to any version deployed

### Checklist
* [x] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [x] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [x] Will you squash when you merge this PR?

💔Thank you!

### CODE REVIEW

File: contracts/VersionResolver.sol [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    function VersionResolver() {
+        owner = msg.sender;
+    }
+
+    // Our resolver supports (supportsInterface & addr) interfaces
+    function supportsInterface(bytes4 interfaceID) constant returns (bool) {
+        return interfaceID == 0x3b3b57de
+            || interfaceID == 0x01ffc9a7;
```
Comment (DJRHails) [inline contracts/VersionResolver.sol]: Anything that doesn't mutate state can be a view function and so gas free.

Comment (JamesDalboth) [reply contracts/VersionResolver.sol]: Ah I copied from a tutorial but yes your right

File: contracts/VersionResolver.sol [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return address(this);
+    }
+
+    function releaseVersion(string version, address addr) only_owner {
+        addresses[version] = addr;
+        addresses['latest'] = addr;
+    }
+
+    function getVersion(string version) constant returns (address) {
+        return addresses[version];
```
Comment (DJRHails) [inline contracts/VersionResolver.sol]: Can do the same for this one?

File: contracts/VersionResolver.sol (hunk 1 above)
Comment (DJRHails) [reply contracts/VersionResolver.sol]: You can do it for more than that function e.g.  getAddress

File: contracts/VersionResolver.sol (hunk 2 above)
Comment (JamesDalboth) [reply contracts/VersionResolver.sol]: yes

**Review by DJRHails — APPROVED** [review]

**Pull request merged**